integration/update_test.go

```diff
@@ -72,3 +72,199 @@ func TestUpdateUpsert(t *testing.T) {
+	}{
+		"DoubleIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(42.13)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(84.26)}},
+		},
+		"DoubleNegativeIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(-42.13)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(0)}},
+		},
+		"DoubleIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", float64(43.13)}},
+		},
+		"DoubleIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", float64(43.13)}},
+		},
+		"DoubleIntIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"DoubleLongIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(43)}},
+		},
+		"IntNegativeIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(-1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(41)}},
+		},
+		"IntIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongNegativeIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(-1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(41)}},
+		},
+		"LongIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"LongIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int64(43)}},
+		},
+
+		"FieldNotExist": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"foo", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(42)}, {"foo", int32(1)}},
+		},
+
+		//"DotNotationInt": {
```

Comment:
may we remove tests as they are not used? 
maybe it's better to create a list of what is necessary to implement for `$inc` in a separate `md` document (and suggest tests there)?

Reply:
Removed commented tests.